### PR TITLE
rest: unify inputs/outputs/code to single dir

### LINKED ISSUE #107: rest: unify inputs/outputs/code into workspace endpoint
Workspace should allow uploads and downloads for any file so we remove the abstraction of code/outputs/code types.

This implies changes in the workflow engines, reana-server and reana-client.

### PR DESCRIPTION
**Pull request by diegodelemos**
* Seed workflow run workspace instead per type (inputs/outputs/code).

Closes #107.
Depends on https://github.com/reanahub/reana-commons/pull/28.

### CODE REVIEW

File: docs/openapi.json [hunk 1]
```diff
@@ -619,8 +619,8 @@
     },
     "/api/workflows/{workflow_id_or_name}/workspace": {
       "get": {
-        "description": "This resource is expecting a workflow UUID and a filename to return its list of code|input|output files.",
-        "operationId": "get_workflow_files",
+        "description": "This resource is expecting a workflow run UUID which workspace file list will be retrieved.",
```
Comment (dinosk) [inline docs/openapi.json]: how about: "This resource retrieves the file list of a workspace, given its workflow run UUID."?

**Comment by dinosk** [comment]
Commenting here, but really this is sth that could be done also in #111, the `experiment/organization` should be removed in: https://github.com/reanahub/reana-workflow-controller/blob/master/reana_workflow_controller/tasks.py#L141

**Comment by diegodelemos** [comment]
Probably  we should stop doing this replace and standardise mounting the shared filesystem at one location ([workflow controller mounts on `/reana`](https://github.com/reanahub/reana-cluster/blob/master/reana_cluster/backends/kubernetes/templates/deployments/workflow-controller-template.yaml#L26) whereas [the rest is doing it on `/data`](https://github.com/reanahub/reana-cluster/blob/master/reana_cluster/backends/kubernetes/templates/deployments/serial-default-worker.yaml#L37)) shall we add it on https://github.com/reanahub/reana-cluster/pull/100 which is related? @dinosk

**Comment by diegodelemos** [comment]
https://github.com/reanahub/reana-commons/pull/28 needs to be merged to have this passing on Travis :)

**Pull request merged**